k8s-dataflow/0.2.0-rc1/docker-entrypoint.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+
+DELAY=3
+THRESHOLD=5
+BUCKET=""
+
+ITER=0
+PARAMS=""
+
+# extract params
+for opt in "$@"; do
+  case "$opt" in
+      --tempLocation=*)
+          BUCKET="${opt#*=}"
+          PARAMS="$PARAMS $opt"
+          shift
+          ;;
+      --gcpTempLocation=*)
+          BUCKET="${opt#*=}"
+          PARAMS="$PARAMS $opt"
+          shift
+          ;;
+      --gcsThreshold=*)
+          THRESHOLD="${opt#*=}"
+          shift
+          ;;
+      --gcsDelay=*)
+          DELAY="${opt#*=}"
+          shift
+          ;;
+      *) # preserve positional arguments
+          PARAMS="$PARAMS $opt"
+          shift
+          ;;
+  esac
+done
+
+if [ -z "$BUCKET" ]; then
+    echo "Missing --tempLocation flag. Exiting."
+    exit 1
+fi
+
+echo "params: ${PARAMS} threshold: ${THRESHOLD} delay: ${DELAY} GOOGLE_APPLICATION_CREDENTIALS: ${GOOGLE_APPLICATION_CREDENTIALS:-Not Available}"
+
+# enable service account if credentials are available
+if [ -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]; then
+  echo "GOOGLE_APPLICATION_CREDENTIALS is not defined, gcloud isn't authenticated!"
+  exit 1
+else
+  gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
+fi
+
+# wait for GCS bucket to be available
+while [ "$ITER" -le "$THRESHOLD" ]; do
+  gsutil ls -b "${BUCKET}"
+  if [ $? -eq 0 ]; then
```

Comment:
Running a command and then checking its exit status `$?` against 0 is not necessary, instead:
```
if ! gsutil ls -b "${BUCKET}"
then
  ...
else
 ...
fi
```

see: https://github.com/koalaman/shellcheck/wiki/SC2181

Reply:
Thanks @peel , that's a great reference, I'll use it from now on! I'll address your feedback as well.

Reply:
Yeah, we probably should lint the scripts with `shellcheck` in CI.